docs/articles/immutability-in-js.md

```diff
@@ -0,0 +1,108 @@
+
+https://gist.github.com/frinyvonnick/98878e45d6a8ccb4618238dbec175ae9
+
+We now have a new object with our new name property and a commonNames property remain unchanged. With this method you can create/overwrite multiple properties at the same time.
+
+### Array.concat
+
+Now let's do it with an array. We want to add two new elements in an array in an immutable way.
+
+https://gist.github.com/frinyvonnick/461d4dbad6bff965a60b734f86619faa
+
+Unlike `Array.push`, `Array.concat` does not mutate our array. Instead, it returns a new array.
+
+https://gist.github.com/frinyvonnick/3959e7972688a41055671bceb6228056
+
+This method is flexible. It takes as many elements as you want. They can be either values or arrays.
+
+### Object.freeze
+
+`Object.freeze` isn't really familiar. It lets you make an object immutable ! It prevents every type of mutation (creation, modification, deletion) induced by the use of setters.
+
+https://gist.github.com/frinyvonnick/d1dd1cbf2924b2428b07713d1ad2adf7
+
+We will try to delete name property after object has been frozen.
+
+https://gist.github.com/frinyvonnick/acc86d1c2fe14c23cd6363946cc253da
+
+Reallocation isn't necessary since the object passed as parameter has been made immutable by `Object.freeze`. This method has two available modes:
+
+- A non-strict mode that does not apply mutations
+- A strict mode that throws a `TypeError` if you try to apply mutations
+
+Be careful, it is not recursive. Our property commonNames isn't immutable.
+
+### Spread operator
+
+The spread operator syntax has been introduced in ES2015 for arrays and in ES2018 for objects. It copies all properties of a given object into a new object literal.
+
+https://gist.github.com/frinyvonnick/61908a0e1bdcd97ba965570796c0c665
+
+With arrays, it copies all values of an array into a new array.
+
+https://gist.github.com/frinyvonnick/7ec4cd039fd79f16864546c51b18ab95
+
+Easily readable and unified between arrays and objects, it replaces nicely assign and concat. It is possible to spread multiple arrays and objects in a same literal.
+
+## Why use immutability?
+
+You found out how to make objects and arrays immutable with JavaScript but we didn't explain yet why using immutability is so necessary nowadays. As developers, we are always looking for a way to write more maintainable and readable code. Some paradigms such as Functionnal Programming are focusing on this.
+
+> Functional programming’s goal is to allow us to think less and write more descriptive code.
+>
+> -- <cite>[Alexander Kondov](https://hackernoon.com/functional-programming-paradigms-in-modern-javascript-immutability-4e9751ca005c)</cite>
+
+It has a declarative approach of programming, which means that you focus on describing what your program must accomplish rather than how it should do it. It gives more meaning to your code so the next developper will understand it more easily. Functional programming brings along other concepts that help reach this goal, such as immutability.
```

Comment:
It gives more meaning to your code so the next developper will understand it more easily. => It gives more meaning to your code so that the next developper can understand it more easily.